// node-ipc/entities/Defaults.js
'use strict';

const os = require('os');

class Defaults {
  constructor() {
    const IPType = os.networkInterfaces()[
      Object.keys(os.networkInterfaces())[0]
      ][0].family;

    this.appspace = 'app.';
    this.socketRoot = '/tmp/';
    this.networkHost = (IPType=='IPv6')? '::1' : '127.0.0.1';
    this.networkPort = 8000;
    this.id = os.hostname();
    this.encoding = 'utf8';
    this.rawBuffer = false;
    this.sync = false;

    Object.defineProperties(
      this,
      {
        silent          : {
          enumerable:true,
          writable:true,
          value:false
        },
        maxConnections  : {
          enumerable:true,
          writable:true,
          value:100
        },
        retry           : {
          enumerable:true,
          writable:true,
          value:500
        },
        maxRetries      : {
          enumerable:true,
          writable:true,
          value:Infinity
        },
        stopRetrying    : {
          enumerable:true,
          writable:true,
          value:false
        },
        IPType          : {
          enumerable:true,
          writable:true,
          value:IPType
        },
        tls             : {
          enumerable:true,
          writable:true,
          value:false
        }
      }
    );
  }
}

module.exports=Defaults;
